pkg/core/consensus/reduction/firststep/reducer.go

```diff
@@ -89,24 +88,22 @@ func (r *Reducer) Finalize() {
 }
 
 // Collect forwards Reduction to the aggregator
-func (r *Reducer) Collect(e consensus.Event) error {
-	ev := reduction.New()
-	if err := reduction.Unmarshal(&e.Payload, ev); err != nil {
-		return err
-	}
+func (r *Reducer) Collect(e consensus.InternalPacket) error {
+	red := e.(message.Reduction)
```

Comment:
`Reduction` messages are unmarshalled and stored in a pointer when received from the network, so this typecast causes a panic. This might be the case for lots of messages out there, so it's probably a good idea to compare the unmarshalling logic with the associated typecast later down the line.